client/src/components/dashboard/Sidebar.tsx

```
@@ -0,0 +1,75 @@
+      {/* Sidebar for Mobile (Slide-in effect) */}
+      <div
+        className={`fixed top-0  left-0 h-full w-64 bg-purple-900 text-white p-4 transform transition-transform ${
+          isOpen ? "translate-x-0" : "-translate-x-full"
+        } md:hidden z-40`}
+      >
+        <button
+          onClick={toggleSidebar}
+          className="absolute top-4 right-4 text-white"
+          aria-label={isOpen ? "Close sidebar" : "Open sidebar"} // For screen readers
+         title={isOpen ? "Close sidebar" : "Open sidebar"} // Tooltip on hover
+        >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)